modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1654,10 +1685,13 @@ private CompletableFuture<Object> processMultiEntryAction(ReadWriteMultiRowRepli
                         }
                     }
 
-                    CompletableFuture<Object> raftFut = rowIdsToDelete.isEmpty() ? completedFuture(null)
-                            : updateAllCommand(request, rowIdsToDelete).thenCompose(this::applyUpdateAllCommand);
+                    if (rowIdsToDelete.isEmpty()) {
+                        return completedFuture(new ReplicaResult(result, null));
```

Comment:
That's a tricky part. We still may want to update safeTime even if there are no rows to delete, otherwise we will need to await idle safe time propagation. We may cover this with separate optimization ticket, but for now I'd prefer calling applyUpdateAllCommand even for empty rows for the simplicity.

Reply:
Not sure if this is actual after merging.